### PR TITLE
Update partitioning docs with tag buckets

### PR DESCRIPTION
**Pull request by sanderson**
This updates partitioning-related documentation with the new "server-side bucketing" (aka tag buckets) functionality introduced in the upcoming release of `influxctl`.

I've only applies changes to the Dedicated docs, but once reviewed, I'll port them over to Clustered.

- [x] Rebased/mergeable

### CODE REVIEW

**Review by jstirnaman — APPROVED** [review]

**Comment by appletreeisyellow** [comment]
cc @domodwyer @savage-engineer @NGA-TRAN @alamb in case you are interested

File: content/influxdb/cloud-dedicated/admin/custom-partitions/define-custom-partitions.md [hunk 1]
```diff
@@ -41,37 +41,48 @@ Use the following command flags to identify
...
+  and number of "buckets" to group tag values into.
+  Provide they tag key and the number of buckets to bucket tag values into
+  separated by a comma: `tagKey,N`.
 - `--template-timeformat`: A [Rust strftime date and time](/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates/#time-part-templates)
   string that specifies the time format in the partition template and determines
   the time interval to partition by.
 
+{{% note %}}
+A partition template can include up to 7 total tag and tag bucket parts
+and only 1 time part.
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/admin/custom-partitions/define-custom-partitions.md:54]: Oh TIL... should I add this to the CLI help output as well?

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md [hunk 2]
```diff
@@ -72,14 +74,15 @@ influxctl database create [flags] <DATABASE_NAME>
...
+| :--- | :---------------------- | :--------------------------------------------------------------------------------------------------------------------------------------- |
+|      | `--retention-period`    | Database retention period (default is 0s or infinite)                                                                                    |
+|      | `--max-tables`          | Maximum tables per database (default is 500, 0 uses default)                                                                             |
+|      | `--max-columns`         | Maximum columns per table (default is 250, 0 uses default)                                                                               |
+|      | `--template-tag`        | Tag to add to partition template (can include multiple of this flag)                                                                     |
+|      | `--template-tag-bucket` | Tag and number of buckets to partition tag values into separated by a comma--for example: `tag1,100` (can include multiple of this flag) |
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md:83]: Are the double hyphen + no spaces intentional?

**Review by powersj — APPROVED** [review]
Thank you again for being on top of this! Two questions in line

File: content/influxdb/cloud-dedicated/admin/custom-partitions/_index.md [hunk 3]
```diff
@@ -217,34 +217,102 @@ production,line=B,station=2 temp=106.5,qty=43i 1704070800000000000
...
-- `B | 1 | 2024-01-01 00:00`
-- `B | 2 | 2024-01-01 00:00`
-- `A | 1 | 2024-01-01 01:00`
-- `A | 2 | 2024-01-01 01:00`
-- `B | 1 | 2024-01-01 01:00`
-- `B | 2 | 2024-01-01 01:00`
+- `A | 0 | 2023-12-31`
+- `B | 0 | 2023-12-31`
+- `A | 0 | 2024-01-01`
+- `B | 0 | 2024-01-01`
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/cloud-dedicated/admin/custom-partitions/_index.md:232]: @appletreeisyellow It's just the way the math worked out. For these two strings, when you hash them and divide by 3, each have the remainder of 0. I included another example below with more buckets so they each end up in a different bucket.

File: content/influxdb/cloud-dedicated/admin/custom-partitions/define-custom-partitions.md (hunk 1 above)
Comment (sanderson) [reply content/influxdb/cloud-dedicated/admin/custom-partitions/define-custom-partitions.md:54]: I wouldn't hurt. If they try to do more than 8 template parts, the server returns an error.

File: content/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates.md [hunk 4]
```diff
@@ -68,6 +80,33 @@ characters must be [percent encoded](https://developer.mozilla.org/en-US/docs/Gl
...
+integer divided by the specified number of buckets:
+
+```rust
+hash(tagValue) % N
+```
+
+Generated partition keys include the unique _tag bucket identifier_ specific to
+each partition.
+
+**Supported number of tag buckets**: 1-100,000
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/cloud-dedicated/admin/custom-partitions/partition-templates.md]: 👍🏻 I just pulled this number from the [source code](https://github.com/influxdata/influxdb_iox/blob/main/data_types/src/partition_template.rs#L282).

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md (hunk 2 above)
Comment (sanderson) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/database/create.md:83]: In the markdown parser we use in the docs, a double-dash renders as an en dash.

- `-` - _(dash)_
- `--` – _(en dash)_
- `---` — _(em dash)_

**Review by appletreeisyellow — APPROVED** [review]

**Comment by sanderson** [comment]
Thanks for the quick review everyone! @powersj, I changed the base branch for this PR to `influxctl-next-version`. I'm going to go ahead and merge it, but it won't go live. I'll ship it with the next `influxctl` release.

**Pull request merged**